tests/unit/store/adapter-interop-test.js

```diff
@@ -277,6 +277,43 @@ test("Find with query calls the correct normalizeResponse", function(assert) {
   assert.equal(callCount, 1, 'normalizeQueryResponse was called');
 });
 
+
+test("Find with query and a response of `false` calls the correct normalizeResponse", function(assert) {
+  var passedQuery = { page: 1 };
+
+  var Person = DS.Model.extend({
+    name: DS.attr('string')
+  });
+
+  var adapter = TestAdapter.extend({
+    query(store, type, query) {
+      return Ember.RSVP.resolve(false);
+    }
+  });
+
+  var callCount = 0;
+
+  var ApplicationSerializer = DS.JSONSerializer.extend({
+    normalizeQueryResponse() {
+      callCount++;
+      return this._super(...arguments);
```

Comment:
Looks like the error is because the `JSONSerializer` doesn't know how to normalize a response of false. Changing this line to `return { data: [] }; ` should fix the test.